streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -68,14 +69,16 @@
      * @throws ProcessorStateException if the base state directory or application state directory does not exist
      *                                 and could not be created
      */
-    public StateDirectory(final String applicationId, final String stateDirConfig, final Time time) {
+    public StateDirectory(final StreamsConfig config,
```

Comment:
I'd prefer this was left as it was. The `StateDirectory` doesn't need a `StreamsConfig`.

Reply:
Same argument as above.